### PR TITLE
Tp export columns

### PR DESCRIPTION
**Pull request by tpoterba**
optionally export a one-line types file

### CODE REVIEW

**Comment by tpoterba** [comment]
jenkins

File: src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala [hunk 1]
```diff
@@ -46,22 +49,36 @@ object ExportGenotypes extends Command {
...
-    val (header, fs) = if (cond.endsWith(".columns")) {
+    val (header, parseResults) = if (cond.endsWith(".columns")) {
       val (h, functions) = ExportTSV.parseColumnsFile(ec, cond, sc.hadoopConfiguration)
       (Some(h), functions)
     }
     else
       Parser.parseExportArgs(cond, ec)
 
-    val a = ec.a
+    Option(options.typesFile).foreach { file =>
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala]: This block is duplicated in exportsamples.

File: src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala [hunk 2]
```diff
@@ -46,22 +49,36 @@ object ExportGenotypes extends Command {
...
+      writeTextFile(file, sc.hadoopConfiguration) { out =>
+        val sb = new StringBuilder
+        header
+          .getOrElse(parseResults.indices.map(i => s"_$i").toArray)
+          .zip(parseResults)
+          .iterator
+          .foreachBetween { case (name, (t, f)) =>
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala]: Can you add an implicit conversion from Array[T] to RichIterable[T]?  Then you can axe the `.iterator`.  (You do this in multiple spots.)

File: src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala (hunk 1 above)
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/ExportGenotypes.scala]: And exportvariants.

**Comment by cseed** [comment]
Few things:

If we get rid of annotatevariants json, how do we deal with _0 problem?  (Related, but not directly, to this PR.)

To use this to export/reimport annotations, it seems not quite trivial to get the annotations back in the right place.  What's the intended command line?  Can we get that as a test case?

Otherwise, back to you.

**Comment by tpoterba** [comment]
It'll look like:
```
exportvariants -o <out> -t <out.types> -c 'v, va'
annotatevariants table <out> --no-header --types @out.types --variant-expr _0 --select _1 --root va
annotatevariants expr -c 'va = va._1'
```

**Comment by tpoterba** [comment]
I don't think we have a _0 problem with annotatevariants json -- previously you'd use `root` to key off of, but now you just use the column identifier (or _0 for a no-header, 1-column file)

**Comment by cseed** [comment]
I meant, where it gets stored, you're left with `va.path.to.json._0.field...`, etc.

**Comment by cseed** [comment]
(1) that leaves `_1` lying around, so you get two copies of the annotations, and (2) you're not worried about how complicated that is?

**Comment by tpoterba** [comment]
The pipeline I posted doesn't leave _1 lying around -- `va = va._1` overwrites `va`

**Comment by tpoterba** [comment]
but still very unclear, I'd agree

**Comment by tpoterba** [comment]
jenkins

**Comment by tpoterba** [comment]
jenkins

**Pull request merged**